### ISSUE TITLE
Type checker error with `actonc build` build order

### ISSUE THREAD

**Issue by ddjoh** [actonlang/acton#956]
### Acton Version

0.11.7.20221007.10.48.11

### Steps to Reproduce

With mod1.act:

```
class Super:
    pass

class Sub(Super):
    def __init__(self):
        pass
```

and mod2.act

```
from mod1 import Super, Sub

def fun() -> Super:
    return Sub()
```

build with `actonc build`

### Expected Behavior

The build to succeed.

### Actual Behavior

We get the error:

```
Building project in /home/ddjoh/workspaces/act-test-1
  Compiling mod1.act for release
   Finished compilation in   0.063 s
  Compiling mod2.act for release

ERROR: Error when compiling mod2 module: Type error
 3:14-18
  |
3 |def fun() -> Super:
  |             ^^^^^
Cannot infer mod1.Sub < mod1.Super
```

We can workaround this by building only mod1.act first and then secondly mod2.act. E.g:

```
$ actonc mod1.act 
Building project in /home/ddjoh/workspaces/act-test-1
  Compiling mod1.act for release
   Finished compilation in   0.060 s
$ actonc mod2.act 
Building project in /home/ddjoh/workspaces/act-test-1
  Compiling mod1.act for release
   Already up to date, in    0.000 s
  Compiling mod2.act for release
   Finished compilation in   0.057 s
```

**Comment by plajjan** [comment]
Isn't this the same or related to #837!?

**Comment by nordlander** [comment]
Seems like its' exactly the same. I already have a fix, but it's buried in the deactorizer branch -- which is COMING UP! :-)

**Comment by plajjan** [comment]
@ddjoh can you check if this bug is still happening on latest main branch now that the new deactorizer is merged?

**Comment by plajjan** [comment]
I have attempted to reproduce this and failed. I think it is fixed with #1020, so closing this.
